hs_core/models.py

```diff
@@ -1791,6 +1792,16 @@ def get_xml(self, pretty_print=True):
         rdfs1_isDefinedBy = etree.SubElement(rdf_Description_resource, '{%s}isDefinedBy' % self.NAMESPACES['rdfs1'])
         rdfs1_isDefinedBy.text = current_site_url() + "/terms"
 
+        # encode extended key/value arbitrary metadata
+        resource = BaseResource.objects.filter(object_id=self.id).first()
+        for key in resource.extra_metadata:
+            hsterms_key_value = etree.SubElement(rdf_Description, '{%s}extendedMetadata' % self.NAMESPACES['hsterms'])
+            hsterms_key_value_rdf_Description = etree.SubElement(hsterms_key_value, '{%s}Description' % self.NAMESPACES['rdf'])
+            hsterms_key = etree.SubElement(hsterms_key_value_rdf_Description, '{%s}key' % self.NAMESPACES['hsterms'])
+            hsterms_key.text = key
+            hsterms_value = etree.SubElement(hsterms_key_value_rdf_Description, '{%s}value' % self.NAMESPACES['hsterms'])
+            hsterms_value.text = resource.extra_metadata[key]
+
```

Comment:
@hyi @pkdash Not sure that's enough structure for an extended element in the long term. 

Even extended metadata elements need -- eventually -- to be subject to some form of value validation. Right now, the "plan" is to put such validation into the metadata element class. It hasn't been done yet -- or done right -- in most cases. 

I need to do some experiments before I can respond further. There are several alternatives built into HSTORE beyond just storing a string value.